pkg/providers/nutanix/reconciler/reconciler_test.go

```diff
@@ -0,0 +1,616 @@
+							APIVersion: "v1",
+							Kind:       "ConfigMap",
+						},
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "test-cm",
+							Namespace: constants.EksaSystemNamespace,
+						},
+					},
+				},
+				Secrets: []*apiv1.Secret{
+					{
+						TypeMeta: metav1.TypeMeta{
+							APIVersion: "v1",
+							Kind:       "Secret",
+						},
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "test-secret",
+							Namespace: constants.EksaSystemNamespace,
+						},
+					},
+				},
+				ClusterResourceSets: []*addonsv1.ClusterResourceSet{},
+			},
+			existingObjs: []client.Object{
+				capiCluster,
+				&apiv1.ConfigMap{
+					TypeMeta: metav1.TypeMeta{
+						APIVersion: "v1",
+						Kind:       "ConfigMap",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-cm",
+						Namespace: constants.EksaSystemNamespace,
+					},
+				},
+				&apiv1.Secret{
+					TypeMeta: metav1.TypeMeta{
+						APIVersion: "v1",
+						Kind:       "Secret",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-secret",
+						Namespace: constants.EksaSystemNamespace,
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "skip when CAPI cluster not found",
+			clusterSpec: &cluster.Spec{
+				Config: &cluster.Config{
+					Cluster: &anywherev1.Cluster{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "non-existent-cluster",
+							Namespace: constants.EksaSystemNamespace,
+						},
+					},
+				},
+			},
+			controlPlane: &nutanix.ControlPlane{
+				ConfigMaps:          []*apiv1.ConfigMap{},
+				Secrets:             []*apiv1.Secret{},
+				ClusterResourceSets: []*addonsv1.ClusterResourceSet{},
+			},
+			existingObjs: []client.Object{},
+			wantErr:      false,
+		},
+		{
+			name: "handle empty control plane resources",
+			clusterSpec: &cluster.Spec{
+				Config: &cluster.Config{
+					Cluster: &anywherev1.Cluster{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "test-cluster",
+							Namespace: constants.EksaSystemNamespace,
+						},
+					},
+				},
+			},
+			controlPlane: &nutanix.ControlPlane{
+				ConfigMaps:          []*apiv1.ConfigMap{},
+				Secrets:             []*apiv1.Secret{},
+				ClusterResourceSets: []*addonsv1.ClusterResourceSet{},
+			},
+			existingObjs: []client.Object{capiCluster},
+			wantErr:      false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fakeClient := fake.NewClientBuilder().
+				WithScheme(scheme).
+				WithObjects(tt.existingObjs...).
+				Build()
+
+			r := &Reconciler{
+				client: fakeClient,
+			}
+
+			logger := test.NewNullLogger()
+			err := r.ensureOwnerReferences(context.TODO(), logger, tt.clusterSpec, tt.controlPlane)
+
+			if tt.wantErr {
+				assert.Error(t, err)
+				if tt.expectedErrMsg != "" {
+					assert.Contains(t, err.Error(), tt.expectedErrMsg)
+				}
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestEnsureOwnerReferencesIntegration(t *testing.T) {
```

Comment:
Can we combine unit test for `ensureOwnerReferences` to avoid redundant testing?